src/test/ui/fn/signature-error-reporting-under-verbose.stderr

```diff
@@ -8,6 +8,7 @@ LL |     needs_ptr(foo);
    |
    = note: expected fn pointer `fn(i32, u32)`
                  found fn item `fn(i32, i32) {foo}`
+   = note: different `fn` items have unique types, even if their signatures are the same
```

Comment:
hm, we really should only be pointing this out for distinct function items that have identical signatures... i can't remember if you had asked me about the behavior is here, but this note seems superfluous.

Reply:
we didn't talk about this specifically, but I probably didn't make the connection somewhere. 
I had this here mainly due to [this comment](https://github.com/rust-lang/rust/issues/102608#issuecomment-1265788858)